src/compiler/scanner.ts

```diff
@@ -832,12 +832,13 @@ namespace ts {
 
     /* @internal */
     export function isIdentifierText(name: string, languageVersion: ScriptTarget | undefined): boolean {
-        if (!isIdentifierStart(name.charCodeAt(0), languageVersion)) {
+        let ch = codePointAt(name, 0);

```

Comment:
if the intention of this function is to verify that the given string can be used as identifier, it should probably also support unicode escape sequences like the Identifier parsing in `scan`

Reply:
`isIdentifierText` should never be fed text containing escape sequences. It operates on values of text, not raw text.